tests/unit/table/test_tensor_table.py

```diff
@@ -0,0 +1,290 @@
+    assert all(table["a"].offsets == tensor_dict["a__offsets"])
+
+
+@pytest.mark.skipif(
+    tf is None, reason="Tensorflow is required for cross-framework validation tests"
+)
+def test_column_type_validation():
+    tensor_dict = {
+        "a__values": np.array([1, 2, 3]),
+        "a__offsets": np.array([0, 1, 3]),
+        "b": tf.constant([4, 5, 6]),
+    }
+
+    with pytest.raises(TypeError) as exc_info:
+        TensorTable(tensor_dict)
+
+    assert "from the same framework" in str(exc_info)
+
+
+@pytest.mark.skipif(
+    tf is None, reason="Tensorflow is required for cross-framework validation tests"
+)
+def test_column_device_validation():
+    with tf.device("/CPU"):
+        tensor_dict_cpu = {
+            "a__values": tf.convert_to_tensor(np.array([1, 2, 3])),
+            "a__offsets": tf.convert_to_tensor(np.array([0, 1, 3])),
+        }
+    with tf.device("/GPU:0"):
+        tensor_dict_gpu = {
+            "b__values": tf.convert_to_tensor(np.array([1, 2, 3])),
+            "b__offsets": tf.convert_to_tensor(np.array([0, 1, 3])),
+        }
+
+    tensor_dict = {**tensor_dict_cpu, **tensor_dict_gpu}
+
+    with pytest.raises(ValueError) as exc_info:
+        TensorTable(tensor_dict)
+
+    assert "on the same device" in str(exc_info)
+
+
+class PaddingOperator(BaseOperator):
+    def __init__(self, length=None, array_lib=np):
+        self.length = length
+        self.array_lib = array_lib
+
+    def transform(self, col_selector, transformable):
+        for col_name, col_data in transformable[col_selector.names].items():
+            # dtype = col_data.dtype.to("numpy")
+
+            dtype = self.array_lib.int32
+            num_rows = len(col_data.offsets) - 1
+            result = self.array_lib.zeros((num_rows, self.length), dtype=dtype)
+
+            for i in range(num_rows):
+                row_length = len(col_data[i])
+                padding = self.array_lib.array([0] * (self.length - row_length), dtype=dtype)
+                padded_row = self.array_lib.append(col_data[i], padding)
+                result[i] = padded_row.astype(dtype)
+            transformable[col_name] = type(col_data)(result)
+
+        return transformable
+
+    # TODO: Define what this op supports (and doesn't)
+
+
+# target input, target column
+# source input, source column
+@pytest.mark.parametrize("source_column", cpu_source_col)
+@pytest.mark.parametrize("target_column", cpu_target_packages)
+def test_tensor_cpu_table_operator(source_column, target_column):
+    source_column_type, source_col_constructor, array_lib = source_column
+    target_column_type, target_input = target_column
+    op = PaddingOperator(3, array_lib=array_lib)
+    expected_output = source_col_constructor([[1, 0, 0], [2, 3, 0]])
+
+    tensor_table = TensorTable(target_input)
+
+    # Column conversions would happen in the executor
+    for col_name, column in tensor_table.items():
+        tensor_table[col_name] = convert_col(column, source_column_type)
```

Comment:
is this checking that converting a column that's already of the target type a no-op? (is this required for the test to pass?

Reply:
It's not required for the test to pass